galsim/phase_screens.py

```diff
@@ -439,33 +439,104 @@ def _zern_rho_coefs(n, m):
     kmax = (n-abs(m)) // 2
     A = [0]*(n+1)
     val = _nCr(n,kmax) # The value for k = 0 in the equation below.
-    norm = _zern_norm(n,m)
     for k in range(kmax):
         # val = (-1)**k * _nCr(n-k, k) * _nCr(n-2*k, kmax-k) / _zern_norm(n, m)
         # The above formula is faster as a recurrence relation:
-        A[n-2*k] = val / norm
+        A[n-2*k] = val
         # Don't use *= since the factor is not an integer, but the result is.
         val = -val * (kmax-k)*(n-kmax-k) // ((n-k)*(k+1))
-    A[n-2*kmax] = val / norm
+    A[n-2*kmax] = val
     return A
 
 
-def _zern_coef_array(n, m, shape=None):
+def _zern_coef_array(n, m, eps=0., shape=None, annular=False):
     """Assemble coefficient array for evaluating Zernike (n, m) as the real part of a
     bivariate polynomial in abs(rho)^2 and rho, where rho is a complex array indicating position on
     a unit disc.
     """
     if shape is None:
         shape = ((n//2)+1, abs(m)+1)
     out = np.zeros(shape, dtype=np.complex128)
-    coefs = np.array(_zern_rho_coefs(n, m), dtype=np.complex128)
+    if annular:
+        coefs = np.array(_annular_zern_rho_coefs(n, m, eps), dtype=np.complex128)
+    else:
+        coefs = np.array(_zern_rho_coefs(n, m), dtype=np.complex128)
+    coefs /= _zern_norm(n, m)
     if m < 0:
         coefs *= -1j
     for i, c in enumerate(coefs[abs(m)::2]):
         out[i, abs(m)] = c
     return out
 
 
+def _binomial(a, b, n):
```

Comment:
Any reason not to put this in utilities.py?  Seems possibly more generally useful, and people would tend to look there for random utilities.